addons/mail/static/src/components/emoji_category/emoji_category.xml

```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<templates xml:space="preserve">
+
+    <t t-name="mail.EmojiCategory" owl="1">
+        <span class="o_EmojiCategory p-1 fs-5 o_cursor_pointer" t-on-click="props.emojiCategoryView.emojiListView.onClickEmojiCategory">
```

Comment:
Define the `t-on` handler on `EmojiCategoryView`, so that you don't rely on `insertAndReplace({ categoryIcon: ev.currentTarget.textContent })` but instead `replace(emojiCategoryView)`.